tf2onnx/onnx_opset/tensor.py

```diff
@@ -1248,3 +1248,142 @@ def version_10(cls, ctx, node, **kwargs):
+                if ax < 0:
+                    axes[i] += len_shape
+            
+            axes = sorted(axes)
+
+            orig_perm = list(range(len_shape))
+            curr_perm = []
+
+            # Add ReverseSequence nodes for each element of axis.
+            for i in range(len_axes):
+
+                axis = axes[i]
+
+                curr_perm = orig_perm.copy()
+                # Permutation indices relative to original tensor.
+                curr_perm[axis], curr_perm[0] = curr_perm[0], curr_perm[axis]
+
+                # Add a Transpose node if the axis != 0 (finish first due to sort).
+                if axis != 0:
+                    # Permutation indices for the transpose node relative to IN tensor shape.
+                    new_node = ctx.make_node(
+                        "Transpose",
+                        inputs=inputs[-1],
+                        op_name_scope=rv2_node_name,
+                        outputs=None,
+                        attr={"perm": curr_perm}
+                    )
+
+                    inputs.append([new_node.output[0]])
+
+                # Add a Constant node (seq_len) for ReverseSequence.
+
+                # Index 1 for the shape should not return 0
+                # since the input must have rank >= 2.
+                rs_batch_size = output_shape[1] if new_node is None \
+                    else ctx.get_shape(new_node.output[0])[1]
+
+                seq_list = [output_shape[axis]] * rs_batch_size
+                seq_array = np.asarray(seq_list)
+
+                const_seq_name = rv2_node_name + '_Const' + str(i)
+                new_node = ctx.make_const(name=const_seq_name, np_val=seq_array)
+                inputs[-1].append(new_node.output[0])
+
+                # Add a ReverseSequence node.
+
+                # If processing for the final axis and the tensor shape permutation is
+                # original then the output is fed to the output of the ReverseV2 node.
+                #
+                # Else a new output is created which is fed to a Transpose node.
+                rs_out_name = rv2_out_name if \
+                    ((i == len_axes - 1) and (curr_perm == orig_perm)) \
+                        else None
+
+                new_node = ctx.make_node(
+                    "ReverseSequence",
+                    inputs=inputs[-1],
+                    op_name_scope=rv2_node_name,
+                    outputs=rs_out_name,
+                    attr={"batch_axis": 1, "time_axis": 0}
+                )
+
+                # Ensure that the dtype of seq_len is INT64.
+                seq_len_dtype = ctx.get_dtype(new_node.input[1])
+                utils.make_sure(seq_len_dtype is not None, "dtype of {} is None".format(node.input[1]))
+                target_dtype = TensorProto.INT64
+                if seq_len_dtype != target_dtype:
+                    ctx.insert_new_node_on_input(new_node, "Cast", new_node.input[1], to=target_dtype)
```

Comment:
no need to check and insert cast here, since seq_len node is created by us so we can control its dtype by
`seq_array = np.asarray(seq_list, dtype=np.int64)`